pymatgen/io/cp2k/inputs.py

```diff
@@ -2732,22 +2724,23 @@ class DataFile(MSONable):
     objects: Sequence | None = None
 
     @classmethod
-    def from_file(cls, filename):
+    def from_file(cls, filename) -> None:
         """Load from a file."""
-        with open(filename) as file:
-            data = cls.from_str(file.read())
-            for obj in data.objects:
-                obj.filename = filename
-            return data
+        raise NotImplementedError
+        # with open(filename, encoding="utf-8") as file:
+        #     data = cls.from_str(file.read())
+        #     for obj in data.objects:
+        #         obj.filename = filename
+        #     return data
```

Comment:
was this change accidental? i believe it breaks the `test_relax_maker` in [`atomate2`](https://github.com/materialsproject/atomate2/blob/563ba2a62c5e8351995e3e775518db7cc7650657/tests/cp2k/jobs/test_core.py#L38)


```py
    response = job.run(store=store)
               ^^^^^^^^^^^^^^^^^^^^
  src/jobflow/core/job.py", line 598, in run
    response = function(*self.function_args, **self.function_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  src/atomate2/cp2k/jobs/core.py", line 402, in make
    return super().make.original(self, structure, prev_dir)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  src/atomate2/cp2k/jobs/base.py", line 176, in make
    task_doc = TaskDocument.from_directory(Path.cwd(), **self.task_document_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  src/atomate2/cp2k/schemas/task.py", line 350, in from_directory
    calc_doc, cp2k_objects = Calculation.from_cp2k_files(
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  src/atomate2/cp2k/schemas/calculation.py", line 375, in from_cp2k_files
    cp2k_objects.update(_get_basis_and_potential_files(dir_name))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  src/atomate2/cp2k/schemas/calculation.py", line 482, in _get_basis_and_potential_files
    data[Cp2kObject.BASIS] = BasisFile.from_file(str(dir_name / "BASIS"))  # type: ignore[index]
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  pymatgen/io/cp2k/inputs.py", line 2754, in from_file
    raise NotImplementedError
NotImplementedError
```

Reply:
Ah! Looks like a huge mistake by me. This certainly is not right. Sorry I don't know why this happened by please revert this.

Reply:
I just had another look: https://github.com/materialsproject/pymatgen/blob/55869a11c6d755e57b38969b34b9b63f0cc257e6/pymatgen/io/cp2k/inputs.py#L2745-L2764

And it doesn't look like intended by me. `from_file` would call `from_str` method which is not implemented, therefore `mypy` would complain about the return type. Therefore I inserted an `NotImplementedError`

Reply:
@janosh @JaGeo  Any comment on this? I'm not seeing much difference from this change. Maybe I'm getting anything wrong?

Reply:
i'm not sure i understand the question. `from_str` can be implemented by a subclass of `DataFile` which would then make `from_file` work correctly if the commented out code is restored

Reply:
You're right. Its child class could overwrite the `from_str` method and make `from_file` work. I would open a PR to fix this now.

Sorry for the trouble cause (and wondering why this is not captured by tests).

Reply:
`pymatgen` has imperfect test coverage. new tests are always appreciated!

Reply:
I guess no project has perfect test 😄 I really wish to help but I don't have any experience with `CP2K`. Sorry.